src/main/java/org/datadog/jmxfetch/JMXAttribute.java

```diff
@@ -295,7 +296,12 @@ private boolean matchBeanRegex(Filter filter, boolean matchIfNoRegex) {
         }
 
         for (Pattern beanRegex : beanRegexes) {
-            if(beanRegex.matcher(beanStringName).matches()) {
+            Matcher m = beanRegex.matcher(beanStringName);
+
+            if(m.matches()) {
+            	for (int i = 0; i<= m.groupCount(); i++) { 
+            		this.beanParameters.put(""+i, m.group(i));
```

Comment:
This is a stylistic issue, not a technical one. I don't love 
```java
""+i
```
I don't think it's very clear. I think it would be better to make it more clear:
```java
Integer.toString(i)
```

Reply:
Good call, that is definitely cleaner. Thanks!